src/DragTextEditor.js

```diff
@@ -403,9 +403,13 @@ childMR=()=>{
   isOtherBorder = () => {
     this.setState({isBorder: false});
   };
-  onText=(text)=>{
-    this.setState({text});
-  }
+  onText = (text) => {
+    this.setState({ text }, () => {
+      if (text && this.props.onTextChanged) {
+        this.props.onTextChanged(this.state.text);
+      }
+    });
+  };
```

Comment:
Can you define `this.props.onTextChanged` just once using;
```
const {
   onTextChanged
  } = this.props;
```
then we can use like this;
```
this.setState({ text }, () => {
    if (text && onTextChanged) {
      onTextChanged(text); // also, we don't need to call state. text parameter will be sufficient.
    }
  });
```